engine/src/main/java/io/camunda/zeebe/engine/state/migration/MemoryBoundedColumnIteration.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.state.migration;
+
+import io.camunda.zeebe.db.ColumnFamily;
+import io.camunda.zeebe.db.DbKey;
+import io.camunda.zeebe.db.DbValue;
+import java.util.function.BiConsumer;
+import org.agrona.collections.MutableLong;
+
+public final class MemoryBoundedColumnIteration {
```

Comment:
Extracted to its own class to make it easier to reuse in all multi tenancy migrations.